File: ios/LivenessExceptions.swift
import ExpoModulesCore

/// Stable error codes returned to the JS layer. Mirror the values in
/// src/types/errors.ts so the `error.code` round-trip is clean across
/// the bridge.
enum LivenessExceptionCode: String {
  case notSupported = "liveness_not_supported"
  case cameraPermissionDenied = "liveness_camera_permission_denied"
  case userCanceled = "liveness_user_canceled"
  case providerError = "liveness_provider_error"
  case invalidBootstrap = "liveness_invalid_bootstrap"
}

final class LivenessException: GenericException<String> {
  override var reason: String {
    return param
  }
}

func livenessError(_ code: LivenessExceptionCode, _ message: String) -> LivenessException {
  return LivenessException("\(code.rawValue)::\(message)")
}

File: android/src/main/java/expo/modules/passkeyliveness/LivenessExceptions.kt
package expo.modules.passkeyliveness

import expo.modules.kotlin.exception.CodedException

class LivenessException(val errorCode: String, message: String) : CodedException(message) {
  override fun getCode(): String = errorCode
}

object LivenessErrorCodes {
  const val NOT_SUPPORTED = "liveness_not_supported"
  const val CAMERA_PERMISSION_DENIED = "liveness_camera_permission_denied"
  const val USER_CANCELED = "liveness_user_canceled"
  const val PROVIDER_ERROR = "liveness_provider_error"
  const val INVALID_BOOTSTRAP = "liveness_invalid_bootstrap"
}
